test/testdata/intrinsics/kernel_raise.rb

```diff
@@ -0,0 +1,79 @@
+# typed: true
+extend T::Sig
+
+class MyError < StandardError
+  extend T::Sig
+
+  sig { params(input: String).void }
+  def initialize(input:)
+    @input = input
+  end
+end
+
+class DefinesToException
+  extend T::Sig
+
+  sig { params(arg0: String).void }
+  def initialize(arg0)
+  end
+
+  sig { returns(Exception) }
+  def exception
+    TypeError.new("hello")
+  end
+end
+
+class DoesNotDefineToException
+  extend T::Sig
+
+  sig { params(arg0: String).void }
+  def initialize(arg0)
+  end
+end
+
+sig {params(cls: T.any(T.class_of(TypeError), T.class_of(MyError))).void}
+def example(cls)
+  0.times do
+    raise MyError
+    #            ^ error: Missing required keyword argument `input` for method `MyError#initialize`
+  end
+
+  0.times do
+    raise cls
+    #        ^ error: Missing required keyword argument `input` for method `MyError#initialize`
+  end
+
+  0.times do
+    raise MyError, "arg"
+    #              ^^^^^ error: Missing required keyword argument `input` for method `MyError#initialize`
+    #              ^^^^^ error: Too many positional arguments provided for method `MyError#initialize`. Expected: `0`, got: `1`
+  end
+
+  0.times do
+    raise cls, "arg"
+    #          ^^^^^ error: Missing required keyword argument `input` for method `MyError#initialize`
+    #          ^^^^^ error: Too many positional arguments provided for method `MyError#initialize`. Expected: `0`, got: `1`
+  end
```

Comment:
WDYT about additional tests:

1. Cases where the `initialize` method takes a single arg, but we provide more to `raise`, e.g. `raise MyError, "arg", ["more", "stuff"]` -- trying to verify that we don't pass those extra arguments to the `new` call.
2. Cases where the `initialize` method takes multiple args, but we only provide the single arg (I think this was the original motivating example?)
3. Cases where the `initialize` method takes multiple args, and we provide those multiple args to `raise`, but of course they are not handled.  This is kind of redundant with case 1, I guess?